### PR TITLE
bugfix(#1638): standardize Transactions

### LINKED ISSUE #1638: Pro mode - Order history seems broken
image

1. The token values are incorrect.
2. The top 2 Open orders are Limit Orders from CoW Swap, would be good to add that info. And all of them are placed through CoW Protocol. The local storage info is shared below:
3. The timestamp shows the time the order will expire, would be good to show the time the order was placed.
4. Icons are missing 😮

### PR DESCRIPTION
**Pull request by adamazad**
# Summary

Fixes #1638 

- Add proper currency logo and formatting

### CODE REVIEW

**Comment by MilanVojnovic95** [comment]
Tested :
The token values are correct ✅ 
The timestamp still shows Expiration Time ❌ 
Most of the icons are still missing ❌ 
I also noticed that the fourth transaction from the list is displayed under the "Unknow" protocol. 🤔 
image

**Review by Mi-Lan — APPROVED** [review]
GJ! Works as expected 
Screenshot 2023-01-03 at 16 10 39

**Comment by 0xVenky** [comment]
> Tested : The token values are correct ✅ The timestamp still shows Expiration Time ❌ Most of the icons are still missing ❌ I also noticed that the fourth transaction from the list is displayed under the "Unknow" protocol. 🤔 image

Hey @MilanVojnovic95  what is the Unknown Protocol here? Can you take a look at the tx and see what protocol this went through?

**Comment by 0xVenky** [comment]
Did some random tests, I think they work perfect for me. Would prefer to change the Expiration time to Created Time for COW so that it is standaradised, but we can leave it for next release if its gonna be more than a few minutes. :)

File: src/pages/Swap/AdvancedTradingViewBox/AdvancedSwapMode/AdvancedSwapMode.component.tsx [hunk 1]
```diff
@@ -68,6 +69,13 @@ export const AdvancedSwapMode = ({ children }: PropsWithChildren) => {
     ? tradesWrapper?.current?.clientHeight - TradesWrapperHeight.PADDING
     : TradesWrapperHeight.MIN
 
+  const allTokens = useAllTokens()
+
+  console.log({
```
Comment (Diogomartf) [inline src/pages/Swap/AdvancedTradingViewBox/AdvancedSwapMode/AdvancedSwapMode.component.tsx:74]: maybe we want to remove the logs

File: src/pages/Swap/AdvancedTradingViewBox/AdvancedSwapMode/OrderHistory/OrderHistoryTransaction/OrderHistoryTransaction.tsx [hunk 2]
```diff
@@ -34,43 +36,75 @@ export const OrderHistoryTransaction = ({ tx }: { tx: Transaction }) => {
...
+  }
+
+  let sellTokenAmount = tx.sellToken?.value
+  let buyTokenAmount = tx.buyToken?.value
+
+  // If the transaction has a creation date and a signature, it's a COW order (safe check)
+  if ('creationDate' in tx && 'signature' in tx) {
+    transactionType = tx.class.toString()
+
+    // Fix the token amounts
```
Comment (Diogomartf) [inline src/pages/Swap/AdvancedTradingViewBox/AdvancedSwapMode/OrderHistory/OrderHistoryTransaction/OrderHistoryTransaction.tsx:71]: it looks like this is more formatting than fixing

```suggestion
    // format the token amounts
```

File: src/pages/Swap/AdvancedTradingViewBox/AdvancedSwapMode/OrderHistory/OrderHistoryTransaction/OrderHistoryTransaction.tsx [hunk 3]
```diff
@@ -34,43 +36,75 @@ export const OrderHistoryTransaction = ({ tx }: { tx: Transaction }) => {
...
+    sellTokenAmount = utils.formatUnits(tx.sellToken?.value, formattedSellToken?.decimals)
+    buyTokenAmount = utils.formatUnits(tx.buyToken?.value, formattedBuyToken?.decimals)
+
+    // Fix the timestamp
```
Comment (Diogomartf) [inline src/pages/Swap/AdvancedTradingViewBox/AdvancedSwapMode/OrderHistory/OrderHistoryTransaction/OrderHistoryTransaction.tsx:75]:
```suggestion
    // Format the timestamp
```

File: src/services/AdvancedTradingView/useAllTrades.hook.ts [hunk 4]
```diff
@@ -60,6 +60,10 @@ export const useAllTrades = (): {
             transaction.status.toUpperCase() !== TransactionStatus.REDEEM
         )
         .splice(0, 5 - pendingTransactions.length)
+        // Sort by conirmation time
```
Comment (Diogomartf) [inline src/services/AdvancedTradingView/useAllTrades.hook.ts:63]:
```suggestion
        // Sort by confirmation time
```

File: src/pages/Swap/AdvancedTradingViewBox/AdvancedSwapMode/OrderHistory/OrderHistoryTransaction/OrderHistoryTransaction.tsx [hunk 5]
```diff
@@ -34,43 +36,75 @@ export const OrderHistoryTransaction = ({ tx }: { tx: Transaction }) => {
...
+  const formattedSellToken = useToken(tx.sellToken.tokenAddress)
+  const formattedBuyToken = useToken(tx.buyToken.tokenAddress)
+
+  // Find the protocol
+  let protocol = 'Unknown'
+  if (isSwapTransaction) {
+    protocol = tx.swapProtocol ?? 'Unknown'
+  } else if (isLimitOrderTransaction) {
+    protocol = 'COW'
+  }
```
[the remark this answers is not in the document]
Comment (adamazad) [reply src/pages/Swap/AdvancedTradingViewBox/AdvancedSwapMode/OrderHistory/OrderHistoryTransaction/OrderHistoryTransaction.tsx:51]: As long as we don't need this logic elsewhere, it's fine to keep it here.

File: src/services/AdvancedTradingView/useAllTrades.hook.ts (hunk 4 above)
Comment (adamazad) [reply src/services/AdvancedTradingView/useAllTrades.hook.ts]: Addressed in https://github.com/ImpeccableHQ/swapr-dapp/pull/1646/commits/83426e029f16a9fc9ce601d3cecea919e55cd64f

File: src/pages/Swap/AdvancedTradingViewBox/AdvancedSwapMode/OrderHistory/OrderHistoryTransaction/OrderHistoryTransaction.tsx (hunk 3 above)
Comment (adamazad) [reply src/pages/Swap/AdvancedTradingViewBox/AdvancedSwapMode/OrderHistory/OrderHistoryTransaction/OrderHistoryTransaction.tsx]: Addressed in https://github.com/ImpeccableHQ/swapr-dapp/pull/1646/commits/83426e029f16a9fc9ce601d3cecea919e55cd64f

File: src/pages/Swap/AdvancedTradingViewBox/AdvancedSwapMode/OrderHistory/OrderHistoryTransaction/OrderHistoryTransaction.tsx (hunk 2 above)
Comment (adamazad) [reply src/pages/Swap/AdvancedTradingViewBox/AdvancedSwapMode/OrderHistory/OrderHistoryTransaction/OrderHistoryTransaction.tsx]: Addressed in https://github.com/ImpeccableHQ/swapr-dapp/pull/1646/commits/83426e029f16a9fc9ce601d3cecea919e55cd64f

File: src/pages/Swap/AdvancedTradingViewBox/AdvancedSwapMode/AdvancedSwapMode.component.tsx (hunk 1 above)
Comment (adamazad) [reply src/pages/Swap/AdvancedTradingViewBox/AdvancedSwapMode/AdvancedSwapMode.component.tsx]: Addressed in https://github.com/ImpeccableHQ/swapr-dapp/pull/1646/commits/83426e029f16a9fc9ce601d3cecea919e55cd64f

**Pull request merged**